### PR TITLE
pkg/lwip: fixes the netif handling for the atwinc15x0 driver

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

With PR #14349 the double stack mode for lwIP was enabled. The ATWINC15x0 driver was provided before the changes in PR #14349 but merged after that. Therefore, the changes for the driver were not taken into account and have to be applied accordingly.

### Testing procedure

Compilation of `tests/lwip` in dual stack mode should succeed.
```
LWIP_IPV4=1 LWIP_IPV6=1 USEMODULE=feather-m0-wifi CFLAGS='-DWIFI_SSID=\"ssid\" -DWIFI_PASS=\"pass\"' \
make BOARD=feather-m0 -C tests/lwip flash
```
If you have the hardware at hand, you can test it with command `ifconfig`.

### Issues/PRs references

Changes for PR #14349

### CODE REVIEW

**Comment by gschorcht** [comment]
@aabadie While testing this PR, I realized that it is necessary to enable module `atwinc15x0` explicitly for `arduino-mkr1000`. Would it make sense to enable `atwinc15x0` as `netdev_default`? What do you think?

**Comment by aabadie** [comment]
> Would it make sense to enable atwinc15x0 as netdev_default? What do you think?

I think it totally makes sense. I thought that was already the case but maybe I missed it in the atwinc15x0 PR.

**Comment by gschorcht** [comment]
> > Would it make sense to enable atwinc15x0 as netdev_default? What do you think?
> 
> I think it totally makes sense. I thought that was already the case but maybe I missed it in the atwinc15x0 PR.

Yes, I also missed it because the two PRs were merged almost simultaneously and there was no conflict that would have pointed that out.

At the moment, Murdock compilations fail sporadically with an error that is not related to the PRs. I had to restart the compilation.

**Comment by gschorcht** [comment]
@benpicco BTW, I have a question related to this PR. At the moment, the `LWIP_NETIF_NUMOF` is hardcoded to 1 or the number of the network devices of only one type. This definition of `LWIP_NETIF_NUMOF` is error prone as well as unsatisfying. 

I recall that we had the same problem with the definition of `GNRC_NETIF_NUMOF` and that there were different proposed approaches. Did we finally solve the problem for `GNRC_NETIF_NUMOF` and if so, how?

**Comment by gschorcht** [comment]
> > Would it make sense to enable atwinc15x0 as netdev_default? What do you think?
> 
> I think it totally makes sense. I thought that was already the case but maybe I missed it in the atwinc15x0 PR.

@aabadie Would it be ok to change it piggybacked with this PR or should I open another one?

**Comment by aabadie** [comment]
I'm fine if it's done here :)

**Comment by gschorcht** [comment]
> I'm fine if it's done here :)

BTW, it was already `netdev_default` but only for `gnrc_netdev_default` :wink:

File: boards/arduino-mkr1000/Makefile.dep [hunk 1]
```diff
@@ -1,6 +1,6 @@
 USEMODULE += boards_common_arduino-mkr
 
-ifneq (,$(filter gnrc_netdev_default,$(USEMODULE)))
+ifneq (,$(filter gnrc_netdev_default netdev_default,$(USEMODULE)))
```
Comment (benpicco) [inline boards/arduino-mkr1000/Makefile.dep:3]: You can drop `gnrc_netdev_default`

Comment (gschorcht) [reply boards/arduino-mkr1000/Makefile.dep:3]: Are you sure, does `gnrc_netdev_default` depend on `netdev_default`

Comment (gschorcht) [reply boards/arduino-mkr1000/Makefile.dep:3]: Changed and squashed it directly.

**Comment by gschorcht** [comment]
@aabadie Thanks for reviewing, testing and merging.

**Pull request merged**